mailipy/gen.py

```diff
@@ -208,6 +209,9 @@ def render_template(template, data):
     print()
     print("Created %d mails in '%s', ready to be sent" % (count, args.outbox))
 
+def convertFrom(s):
+    m = re.match(r'(?P<name>[^\<]*?)<(?P<mail>.*)?>', s)
+    return ("=?utf-8?b?%s?=<" + m['mail'] + ">") % (base64.b64encode(m['name'].encode('utf-8')).decode('utf-8'),)
```

Comment:
I think it's not required that the from string follows the `s <s@s>` format, it could also just be `s@s` I think. In such a case I guess the match function would fail. (But maybe I remember this wrong...)

Also: what does this do exactly? Can we somehow encode the whole string in utf-8? If not, would it be better if we start asking for the from_name and from_email separately in the `*.md` template header?

Reply:
I didn't know about formataddr, it could be what we need.

Maybe we can use parsaddr as well, which would work well with formataddr?

```
>>> import email.utils as e
>>> e.parseaddr("實驗 <實驗@實驗.com>")
('實驗', '實驗@實驗.com')
>>> e.parseaddr("<實驗@實驗.com>")
('', '實驗@實驗.com')
>>> e.parseaddr("實驗@實驗.com")
('', '實驗@實驗.com')
```

Does it look ok?

Reply:
I use  re to split name and email because I don't want to change .md.
if we think the script will support chinese name character , maybe use  from_name and from_email separately is another way.
shall we think about compatible old .md files?

Reply:
Could you send me a small .md file which reproduces the bug with send header? You can upload it here or send it to williamdiluigi@gmail.com

Reply:
---

from:     "中文发件人 <noreply@itacpc.it>"
to:       ["{{Email}}"]
cc:       []
bcc:      []
subject:  "中文测试邮件"
msgid:    "<%s@itacpc.it>"
attach:   []
images:
  - path: "./logo.png"
    cid: 0

---

![](cid:0)

Salve {{Titolo}} {{Cognome}},
中文
Con la presente email annunciamo la data del prossimo round di allenamento (o
"warm-up round") della ITACPC. Si svolgerà **sabato 9 novembre** dalle ore
**14:00** alle **18:00**.

Ricordiamo che la gara si svolge online. Gli studenti possono partecipare
individualmente o in squadre.

Saremmo felici se {{ArticoloUniversità}}{{Università}} prendesse parte a questa
gara!

- Le iscrizioni per gli studenti della sua università sono accessibili alla
  pagina <https://teams.itacpc.it/{{Uni}}>
- I problemi sono in lingua inglese. Maggiori informazioni si potranno ottenere
  sul [sito della competizione](https://itacpc.it).
- La durata è di **4 ore**.
asdf十点钟
Per qualsiasi domanda può contattarci.

Cordiali Saluti,  
Lo Staff delle ITACPC.